File: tests/fixtures/safe_secret_names.java
// Benign names containing secret-keyword substrings, or low-signal
// keywords with env-sourced values. None should be flagged by
// java/no-hardcoded-secret after the word-boundary + value-gate fix.
public class SafeSecretNames {
    // Substring false positives.
    static final String author = "Pallets";
    static final String authors = "core team";
    static final String authenticated = "yes";
    static final String authorizationScheme = "Bearer";
    static final String tokenizer = "bert-base-uncased";
    static final String secretarialNote = "filed";

    static void load() {
        // Low-signal + secret-named values, all env-sourced (not literals).
        String auth = System.getenv("AUTH");
        String token = System.getenv("TOKEN");
        String password = System.getenv("PW");
        String apiKey = System.getenv("API_KEY");
        String secretKey = System.getenv("SECRET_KEY");
        System.out.println(auth + token + password + apiKey + secretKey);
    }
}

File: tests/fixtures/safe_secret_names.kt
// Benign names containing secret-keyword substrings, or low-signal
// keywords with env-sourced values. None should be flagged by
// kt/no-hardcoded-secret after the word-boundary + value-gate fix.
object SafeSecretNames {
    // Substring false positives.
    val author = "Pallets"
    val authors = "core team"
    val authenticated = "yes"
    val authorizationScheme = "Bearer"
    val tokenizer = "bert-base-uncased"
    val secretarialNote = "filed"

    fun load() {
        // Low-signal + secret-named values, all env-sourced (not literals).
        val auth = System.getenv("AUTH")
        val token = System.getenv("TOKEN")
        val password = System.getenv("PW")
        val apiKey = System.getenv("API_KEY")
        val secretKey = System.getenv("SECRET_KEY")
        println(auth + token + password + apiKey + secretKey)
    }
}
